girder_worker/plugins/docker/executor.py

```diff
@@ -163,7 +164,7 @@ def make_pipe(id, spec, bindings):
 
 def _run_container(image, args, **kwargs):
     # TODO we could allow configuration of non default socket
-    client = docker.from_env(version='auto')
+    client = nvidia.get_client_for_image(image)
```

Comment:
This is surprising to read.  We are in _run_container, then we call nvidia.get_client_for_image whether I'm using an nvidia Docker or not.  Maybe a comment here or a different method name than get_client_for_image would make this more clear, that I'm only getting nvidia client when appropriate.  

or maybe something like 

```
client = DockerClient.from_env(version='auto')
if nvidia.is_nvidia_image(client.api, image):
  client = nvidia.get_nvidia_client()
```